lib/Services/RetrieveMetadata.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Services;
+
+
+use OCP\IDBConnection;
+
+class RetrieveMetadata {
+
+	protected $dbConnection;
+
+	/**
+	 * @param IDBConnection $connection
+	 */
+	public function __contruct(IDBConnection $connection){
+		$this->dbConnection = $connection;
+	}
+
+	/**
+	 * @param $fileList : List of File IDs
+	 * @return array
+	 */
+	public function retrieve($fileList) {
+
+		$retrievedData = array();
+		$mapper = new ImageDimensionMapper($this->dbConnection);
+
+		$logger = \OC::$server->getLogger();
+
+		foreach ($fileList as $fileID) {
+			$entity = $mapper->find($fileID);
```

Comment:
Should I test the mapper->find() seperately? But how could I test retrieve because in this way I will not be able to control what happens when RetrieveMetadata->retrieve is called at the end of test, right?